lib/api/objectCopy.js

```diff
@@ -55,6 +55,12 @@ function _prepMetadata(sourceObjMD, headers, sourceIsDestination, authInfo,
     const userMetadata = whichMetadata === 'COPY' ?
         utils.getMetaHeaders(sourceObjMD) :
         utils.getMetaHeaders(headers);
+
+    // If COPY, pull AWS-compatible headers from source object
+    // Otherwise, pull AWS-compative headers from request headers
```

Comment:
typo here in compatible.  In any case, instead of compatHeadersSource I would call these headersToStoreSource.  Everything we do is compatible 😄